### DISCUSSION TITLE
(Unknown ??) OS value for UA strings

### DISCUSSION THREAD

**Discussion by nvysyaraju-sift** [nielsbasjes/yauaa#976] [Q&A] [answered]
Hello

While parsing UA strings with the Yauaa parser and observing OS attribute values I found that for certain type of strings with Android OS, the OS value for `OPERATING_SYSTEM_NAME_VERSION_MAJOR` is returned as **Unknown ??**. I understand that these does not fall into a traditional UA string format but just wondering is it possible to get these fixed to extract proper OS value for these or is it better to ignore these are cases with noise or non-traditional string format? Thank you.

```
uphold-sdk-javascript/v2.4.1 (taro; 12; en_US) Wallet Android/4.79.1

Vestiaire Collective/3.92.0 (LenovoTB-X606X_EEA:mt8768 Android:29)

Deposit Forwarded - Vestiaire Collective/3.93.0 (c2sxxx:exynos990 Android:31)

member-react-native-application/android/1.11.12/6562/47b4b800cb7e06905a51a53ab8c05a88
```

**Comment by nielsbasjes** [comment, marked as answer]
Yes, these are indeed very uncommon useragents.
I'll have to examine them more to know if any detection can be done.
Do you have more examples like these?

**Reply by nvysyaraju-sift** [reply]
From what I've seen those are the different string formats that seem to be returning **Unknown ??** and have android in their strings. The above listed strings are not the only strings but are the string formats.

**Reply by nielsbasjes** [reply]
Fixed. :)
